### PR TITLE
[SM-455] add projects and secrets to dashboard page

### PR DESCRIPTION
**Pull request by willmartian**
## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR adds a collapsible secrets table and projects table to the SM dashboard page. 

## Code changes

- `secrets-list` and `projects-list` now take a `basePath` Input to deal with any relative routerLinks. 
- `secrets-list` now uses `bitSortable`
- `sm-overview` displays the projects and secrets lists
- A collapsible section component was created, `sm-section`.
- The org title was added to the `sm-header` of `sm-overview`
- `bitSortable`styles were updated to prevent wrapping the sort icon to a new line

## Screenshots

https://user-images.githubusercontent.com/17113462/218877728-42d43fa9-59a7-4b21-ad69-b5ba098b1793.mov

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: libs/components/src/table/sortable.component.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ import { TableComponent } from "./table.component";
   selector: "th[bitSortable]",
   template: `
     <button
-      class="tw-group"
+      class="tw-group tw-min-w-max"
```
Comment (Hinton) [inline libs/components/src/table/sortable.component.ts:11]: Any reason we're using both ngClass and class? We could lift these into the classList below.

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts [hunk 2]
```diff
@@ -13,6 +13,8 @@ import { ProjectListView } from "../../models/view/project-list.view";
 export class ProjectsListComponent implements OnDestroy {
   protected dataSource = new TableDataSource<ProjectListView>();
 
+  @Input() baseRoute = ".";
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts:16]: I wonder if this component should fetch the org id from the route and use an absolute url instead. Not really happy with either option.

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.module.ts [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
 import { SecretsManagerSharedModule } from "../shared/sm-shared.module";
 
 import { OverviewRoutingModule } from "./overview-routing.module";
 import { OverviewComponent } from "./overview.component";
+import { SectionComponent } from "./section.component";
 
 @NgModule({
-  imports: [SecretsManagerSharedModule, OverviewRoutingModule],
-  declarations: [OverviewComponent],
+  imports: [SecretsManagerSharedModule, OverviewRoutingModule, ProjectsModule, SecretsModule],
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.module.ts:12]: Can we just have these modules export their list component?

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts [hunk 4]
```diff
@@ -1,7 +1,179 @@
...
+      .pipe(
+        map((params) => this.organizationService.get(params.organizationId)),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((org) => {
+        this.organizationId = org.id;
+        this.organizationName = org.name;
+      });
+
+    this.projects$ = this.projectService.project$.pipe(
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts:77]: Change to a local variable?

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts [hunk 5]
```diff
@@ -1,7 +1,179 @@
...
+      startWith(null),
+      combineLatestWith(this.route.params),
+      switchMap(() => this.getProjects())
+    );
+
+    this.secrets$ = this.secretService.secret$.pipe(
```
Comment (Hinton) [inline bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.component.ts:83]: Local variable?

File: bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.module.ts (hunk 3 above)
Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/overview/overview.module.ts:12]: These are no longer needed; left over code from a prior approach earlier in this PR. Removed 👍 

The lists are declared in the `SecretsManagerSharedModule`

File: bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts (hunk 2 above)
Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts]: @Hinton I removed `basePath` and added a `viewProjectEvent` to handle the navigation instead. Thoughts? [87ccfbe](https://github.com/bitwarden/clients/pull/4722/commits/87ccfbe34fff8b39572842a0a88c916d110e5b74)

Comment (Hinton) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts]: That thought occurred to me but I'm not a huge fan of non anchors that acts as links.

Comment (willmartian) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts]: Okay, now we are using an absolute route with the org id that is provided in the `ProjectListView`. [ecfc4ce](https://github.com/bitwarden/clients/pull/4722/commits/ecfc4ce99766e6fcda894498321dcd077b8f41d6)

Comment (Hinton) [reply bitwarden_license/bit-web/src/app/secrets-manager/shared/projects-list.component.ts]: I still hate it, but that's probably the best solution 😢. Thanks 👍

**Review by Hinton — APPROVED** [review]

**Pull request merged**